### DISCUSSION TITLE
Feature Request - Add a way to only copy some data when comparing json files

### DISCUSSION THREAD

**Discussion by xdeiimoss** [josdejong/svelte-jsoneditor#503] [Ideas]
An option should be added when comparing two json files. One that would allow users to copy only selected keys to the other panel.
For example, I know that I have some data on the left panel that I wouldn't like to have in the right panel, so I select the keys that I want to copy.

Another good thing that is related to this is basically adding a way to just add a list of keys that you want to be copied from one to another. In this example, I would like to have the viewerFlags, history, lastModifiedAt, favoriteModifiedAt, version and customStatus
![image](https://github.com/user-attachments/assets/a98bf0a5-2db0-4222-8921-4fcc1b7483ea)

Would this be possible?
Thank you in advance

**Comment by josdejong** [comment]
Moving this idea to the discussions section, let's discuss there further.

**Comment by josdejong** [comment]
It's interesting to think this through, though I'm not sure there is a quick solution to this. I think it also depends on your use case.

1. One use case could be comparing two versions of a document and wanting to pick only a selection of the changes, like with a git diff.
2. Another use case is more generic: implement support for multi selection, so you can copy/paste that. See https://github.com/josdejong/svelte-jsoneditor/discussions/356
3. If this is a matter of a repetitive action, it may be interesting to write a Transform query to transform from one to the other.

Just thinking aloud here :)

**Reply by xdeiimoss** [reply]
I feel like the 1st option would be best in my case personally...

**Reply by AlexRMU** [reply]
I think both the first and the second, and also add the "select added/deleted/unchanged" buttons to the menu, create and select hunks.
